tests/integration/models/alert_test.py

```diff
@@ -0,0 +1,162 @@
+from datetime import datetime
+import logging
+from mock import patch
+
+from nav.alertengine.base import clear_blacklisted_status_of_alert_senders
+from nav.alertengine.dispatchers import InvalidAlertAddressError
+from nav.models.profiles import (
+    Account,
+    AccountAlertQueue,
+    AlertAddress,
+    AlertProfile,
+    AlertSender,
+    AlertSubscription,
+    FilterGroup,
+    TimePeriod,
+)
+from nav.models.event import AlertQueue, Subsystem
+
+import pytest
+
+
+def test_delete_alert_subscription(db, alert, alertsub, account_alert_queue):
+    alertsub.delete()
+    assert not AccountAlertQueue.objects.filter(pk=account_alert_queue.pk).exists()
+    assert not AlertQueue.objects.filter(pk=alert.pk).exists()
+
+
+def test_sending_alert_to_alert_address_with_invalid_address_will_raise_error(
+    db, alert_address, alert, alertsub
+):
+    with pytest.raises(InvalidAlertAddressError):
+        alert_address.send(alert, alertsub)
+
+
+def test_sending_alert_to_alert_address_with_invalid_address_will_delete_alert_and_fail(
+    db, alert, account_alert_queue
+):
+    assert not account_alert_queue.send()
+    assert not AlertQueue.objects.filter(pk=alert.pk).exists()
+
+
+def test_sending_alert_via_blacklisted_sender_will_fail_but_not_delete_alert(
+    alert, alert_address, account_alert_queue, caplog
+):
+    alert_address.address = "47474747"
+    alert_address.save()
+    alert_address.type.blacklisted_reason = "This has been blacklisted because of x."
+    alert_address.type.save()
+    with caplog.at_level(logging.DEBUG):
```

Comment:
```suggestion
    with caplog.at_level(logging.DEBUG, logger="nav.alertengine.alertaddress.send"):
```

A problem with caplog is that it attaches to the `root` logger.  If, for some reason, a logger in the hierarchy is set to *not* propagate log messages upwards in the hierarchy, caplog will never see the message.

I think your caplog problem can be solved by attaching directly to the expected logger.

*However*, I don't like tests that make assertions about log output, see my other comments.